### PR TITLE
Create a cleanup handler for the varnish cache to provide 'ban' functionality

### LINKED ISSUE #3: Clear varnish cache for an asset
a dependency for https://github.com/dlcs/protagonist/issues/251

## Acceptance criteria

- [ ] A side-process that listens to the same Asset Deletion notification as https://github.com/dlcs/protagonist/issues/251
- [ ] When it receives an asset deletion message, uses varnishadm BAN command to clear matching stored entries. e.g., `ban req.url ~ /iiif-img/wellcome/5/b33062055`
- [ ] iiif-img, iiif-manifest, thumbs, file, iiif-av are "wildcard" paths to clear matching asset IDs under (single assets)

### PR DESCRIPTION
**Pull request by JackLewis-digirati**
Resolves #3 

This pull request adds a python script that listens to an SQS queue for deletions and then actions a `ban` request on a varnish cache using tags.   This script can be run separately, in it's own docker container, or within the varnish container itself

### CODE REVIEW

File: .github/actions/docker-build-and-push/action.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        images: ghcr.io/dlcs/${{ inputs.image-name }}
+        tags: |
+          type=ref,event=branch
+          type=ref,event=pr
+          type=sha,enable=true,prefix=,format=long
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+    - id: docker-login
+      uses: docker/login-action@v1
```
Comment (donaldgray) [inline .github/actions/docker-build-and-push/action.yml:40]: `docker/login-action@v2`

File: .github/actions/docker-build-and-push/action.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ inputs.github-token }}
+    - id: docker-build-push
+      uses: docker/build-push-action@v2
```
Comment (donaldgray) [inline .github/actions/docker-build-and-push/action.yml:46]: `docker/build-push-action@v4`

File: .github/actions/docker-build-and-push/action.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  steps:
+    - id: checkout
+      uses: actions/checkout@v2
+    - id: docker-setup-buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        driver-opts: |
+          image=moby/buildkit:v0.10.6
+    - id: docker-meta
+      uses: docker/metadata-action@v3
```
Comment (donaldgray) [inline .github/actions/docker-build-and-push/action.yml:29]: `docker/metadata-action@v4`

File: .github/actions/docker-build-and-push/action.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+runs:
+  using: "composite"
+  steps:
+    - id: checkout
+      uses: actions/checkout@v2
+    - id: docker-setup-buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        driver-opts: |
+          image=moby/buildkit:v0.10.6
```
Comment (donaldgray) [inline .github/actions/docker-build-and-push/action.yml:27]: I'm not sure if this is required any longer, I'll take a note to test removing from Protagonist. It was related to a ghcr/buildkit issue from a few months ago (https://github.com/docker/build-push-action/issues/761#issuecomment-1398918693)

File: varnish-cleanup/app/aws_factory.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import boto3
+from moto import sqs
+
+from app.settings import LOCALSTACK, REGION, LOCALSTACK_ADDRESS
+from logzero import logger
+
+
+def get_aws_client(client_type: str):
```
[the remark this answers is not in the document]
Comment (JackLewis-digirati) [reply varnish-cleanup/app/aws_factory.py]: removed

**Pull request merged**